spread.yaml

```diff
@@ -0,0 +1,108 @@
+project: zinc-charm-tests
+
+environment:
+  JUJU_CHANNEL/juju_3_5: 3.5/stable
+  JUJU_CHANNEL/juju_4_0: 4.0/beta
+  CHARMCRAFT_CHANNEL: latest/stable
+  LXD_CHANNEL: latest/stable
+  MICROK8S_CHANNEL: 1.31-strict/stable
+  PROVIDER: microk8s
+
+  MICROK8S_ADDONS: hostpath-storage dns
+  METALLB_RANGE: 10.64.140.43-10.64.140.49
+
+  JUJU_BOOTSTRAP_OPTIONS: --model-default test-mode=true --model-default automatically-retry-hooks=false
+  JUJU_EXTRA_BOOTSTRAP_OPTIONS: ""
+  JUJU_BOOTSTRAP_CONSTRAINTS: ""
+
+  LANG: "C.UTF-8"
+  LANGUAGE: "en"
+
+  PROJECT_PATH: /home/spread/proj
+  CRAFT_TEST_PATH: /home/spread/proj/tests/spread/lib
+
+backends:
+  lxd:
+    type: adhoc
+    allocate: |
+      BASE="${BASE:-noble}"
+      VM_NAME="${VM_NAME:-ubuntu-${BASE}-${RANDOM}}"
+      DISK="${DISK:-20}"
+      CPU="${CPU:-4}"
+      MEM="${MEM:-8}"
+
+      lxc launch --vm \
+        "ubuntu:${BASE}" \
+        "${VM_NAME}" \
+        -c user.user-data="$(cat tests/spread/lib/cloud-config.yaml)" \
+        -c limits.cpu="${CPU}" \
+        -c limits.memory="${MEM}GiB" \
+        -d root,size="${DISK}GiB"
+
+      # Wait for the spread user
+      while ! lxc exec "${VM_NAME}" -- id -u spread &>/dev/null; do sleep 0.5; done
+
+      # Set the instance address for spread
+      ADDRESS "$(lxc ls -f csv | grep "${VM_NAME}" | cut -d"," -f3 | cut -d" " -f1)"
+    discard: |
+      instance_name="$(lxc ls -f csv | grep $SPREAD_SYSTEM_ADDRESS | cut -f1 -d",")"
+      lxc delete -f $instance_name
+
+    systems:
+      - ubuntu-24.04:
+          username: spread
+          password: spread
+          workers: 1
+
+  local:
```

Comment:
If you rename this to `github-ci` you can simply run `charmcraft test` in the CI workflow.

Reply:
Yeh I saw that in the code -- the only reason it might not make much difference is because I need to explicitly call the `local`/`github-ci` backend jobs, and *not* the `lxd` jobs, so either way I need to specify a backend I guess